commcare_connect/opportunity/views.py

```diff
@@ -73,7 +73,18 @@ class OpportunityList(OrganizationUserMixin, ListView):
     paginate_by = 10
 
     def get_queryset(self):
-        return Opportunity.objects.filter(organization=self.request.org)
+        return (
+            Opportunity.objects.filter(organization=self.request.org)
+            .annotate(
+                is_active_val=Case(
+                    When(Q(end_date__gte=now()) & Q(active=True), then=True),
+                    default=False,
+                    output_field=BooleanField(),
+                )
+            )
+            .order_by("name")
+            .order_by("-is_active_val")
```

Comment:
Is this sort name then active? If so, I think that won't do quite the right thing. I think we want the opposite order of terms. For example if there are three opportunities A, B, and C and C is inactive, we want the order to be A, C, B. I think this would order those A, B, C since activity would tiebreak name order rather than names tiebreaking activity. Let me know if I am misinterpreting how this code works. Otherwise looks good, although I forgot that active wasn't part of the model and would require an annotation, so if you would rather, I think it is fine to just order by name.

Reply:
the order currently is sort name then active. That ensures that opportunities that are active are displayed before the opportunities which are inactive and are sorted alphabetically, as you said in the earlier comment. I am happy to make this ordered by name only, as it would remove this annotation which adds computational overhead to the database.

Reply:
I poked around in a shell and I don't think this works. It looks like using `order_by` twice like you are doing just overwrites the first with the second, so this is only sorting by `is_active_val`. You can verify this with `print(<my_orm_query>.query)`. I would be fine changing this to just name, otherwise you can sort on two values, by passing them both as arguments to ordery_by like `order_by("-is_active_val", "name")`

Reply:
Thanks that makes sense. I will make the change to sort this only by `name`.

Reply:
Fixed. https://github.com/dimagi/commcare-connect/pull/233/commits/da61df23463d1eaacf7675b9a7b3388cd40d0f1a